### PR TITLE
AWS cloudformation policies review

### PR DESCRIPTION
**Pull request by mr-maxo**

### CODE REVIEW

File: aws/iac/dynamodb.rego [hunk 1]
```diff
@@ -8,40 +8,37 @@ package rule
...
     lower(resource.Type) == "aws::dynamodb::table"
-    not resource.Properties.SSESpecification.SSEType
+    not resource.Properties.SSESpecification.SSEEnabled
 }
 
 aws_issue["dynamodb_encrypt"] {
     resource := input.Resources[i]
     lower(resource.Type) == "aws::dynamodb::table"
-    lower(resource.Properties.SSESpecification.SSEType) == "aes256"
+    resource.Properties.SSESpecification.SSEEnabled != true
```
Comment (harshilswan) [inline aws/iac/dynamodb.rego:26]: This is not covering string value "false" scenario. please refer [ec2.rego](https://github.com/prancer-io/prancer-compliance-test/blob/master/aws/iac/ec2.rego#L90) for checking bool values

File: aws/iac/ebs.rego [hunk 2]
```diff
@@ -11,7 +11,7 @@ default ebs_encrypt = null
 aws_issue["ebs_encrypt"] {
     resource := input.Resources[i]
     lower(resource.Type) == "aws::ec2::volume"
-    not resource.Properties.Encrypted
+    resource.Properties.Encrypted != true
```
Comment (harshilswan) [inline aws/iac/ebs.rego:14]: This is not covering Encrypted is not defined or "bool value as string" scenario. please refer [cloudtrail.rego](https://github.com/prancer-io/prancer-compliance-test/blob/master/aws/iac/cloudtrail.rego#L9).

File: aws/iac/dynamodb.rego (hunk 1 above)
Comment (mr-maxo) [reply aws/iac/dynamodb.rego:26]: Fix: git show c0f6e4c

File: aws/iac/ebs.rego (hunk 2 above)
Comment (mr-maxo) [reply aws/iac/ebs.rego:14]: Fix: `git show e1d9096`

File: aws/iac/ec2.rego [hunk 3]
```diff
@@ -92,7 +92,9 @@ default ec2_public_ip = null
 aws_issue["ec2_public_ip"] {
     resource := input.Resources[i]
     lower(resource.Type) == "aws::ec2::instance"
-    resource.Properties.NetworkInterfaces[_].AssociatePublicIpAddress == true
+    nic := resource.Properties.NetworkInterfaces[_]
+    nic.AssociatePublicIpAddress == true
+    lower(resource.SecurityGroups[_]) == "default"
```
Comment (harshilswan) [inline aws/iac/ec2.rego:103]: will be like resource.Properties.SecurityGroups[] not resource.SecurityGroups[]

Comment (mr-maxo) [reply aws/iac/ec2.rego:103]: Yes, fixed

File: aws/iac/dynamodb.rego (hunk 1 above)
Comment (mr-maxo) [reply aws/iac/dynamodb.rego:26]: Added string check, fixed

File: aws/iac/ebs.rego (hunk 2 above)
Comment (mr-maxo) [reply aws/iac/ebs.rego:15]: added string check, fixed

File: aws/iac/ec2.rego (hunk 3 above)
Comment (harshilswan) [reply aws/iac/ec2.rego:103]: this line should be like `lower(resource.Properties.SecurityGroups[_]) == "default"`

**Review by harshilswan — APPROVED** [review]

**Pull request merged**